core/src/main/java/org/apache/brooklyn/core/mgmt/internal/BrooklynGarbageCollector.java

```diff
@@ -92,16 +93,28 @@
             Duration.minutes(1));
     
     public static final ConfigKey<Boolean> DO_SYSTEM_GC = ConfigKeys.newBooleanConfigKey(
-            "brooklyn.gc.doSystemGc", "whether to periodically call System.gc()", false);
-    
+        "brooklyn.gc.doSystemGc", "whether to periodically call System.gc()", false);
+
+    public static final ConfigKey<Double> FORCE_CLEAR_SOFT_REFERENCES_ON_MEMORY_USAGE_LEVEL = 
+        ConfigKeys.newDoubleConfigKey("brooklyn.gc.clearSoftReferencesOnMemoryUsageLevel", 
+            "force clearance of soft references (by generating a deliberate OOME) "
+            + "if memory usage gets higher than this percentage of available memory; "
+            + "Brooklyn will use up to the max, or this percentage, with soft references,"
+            + "so if using any high-memory-usage alerts they should be pegged quite a bit"
+            + "higher than this threshhold "
+            + "(default >1 means never)", 2.0);
```

Comment:
The default an unusual way of saying never. `null` is the non-surprising value for this case, even if it complicates the code a bit.